grammars/css.cson

```diff
@@ -20,294 +20,1218 @@
     'include': '#comment-block'
   }
   {
+    'include': '#escapes'
+  }
+  {
+    'include': '#combinators'
+  }
+  {
     'include': '#selector'
   }
   {
-    'begin': '(@)charset\\b'
-    'captures':
+    'include': '#at-rules'
+  }
+  {
+    'include': '#rule-list'
+  }
+  {
+    'match': ';'
+    'name': 'punctuation.terminator.rule.css'
+  }
+  {
+    'match': '}'
+    'name': 'punctuation.section.end.css'
+  }
+]
+'repository':
+  'at-rules':
+    'patterns': [
+      {
+        # @charset
+        'begin': '\\A(?:\\xEF\\xBB\\xBF)?(?i:(?=\\s*@charset\\b))'
+        'end': ';|(?=$)'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.terminator.rule.css'
+        'name': 'meta.at-rule.charset.css'
+        'patterns': [
+          {
+            'captures':
+              '1':
+                'name': 'invalid.illegal.not-lowercase.charset.css'
+              '2':
+                'name': 'invalid.illegal.leading-whitespace.charset.css'
+              '3':
+                'name': 'invalid.illegal.no-whitespace.charset.css'
+              '4':
+                'name': 'invalid.illegal.whitespace.charset.css'
+              '5':
+                'name': 'invalid.illegal.unquoted-identifier.charset.css'
+              '6':
+                'name': 'invalid.illegal.unclosed-string.charset.css'
+              '7':
+                'name': 'invalid.illegal.unexpected-characters.charset.css'
+            'match': '''(?x)        # Possible errors:
+              \\G
+              ((?!@charset)@\\w+)   # Not lowercase (@charset is case-sensitive)
+              |
+              \\G(\\s+)             # Preceding whitespace
+              |
+              (@charset\\S[^;]*)    # No whitespace after @charset
+              |
+              (?<=@charset)         # Before quoted charset name
+              (\\x20{2,}|\\t+)      # More than one space used, or a tab
+              |
+              (?<=@charset\\x20)    # Beginning of charset name
+              ([^"';]+)             # Not quoted
```

Comment:
If you could upload some images of the behavior you're seeing just so that I can verify it's the same as what is displaying on my end, that would be appreciated.  It still appears to me that the ISO example isn't using the correct encoding with single quotes.

![single-quotes](https://cloud.githubusercontent.com/assets/2766036/19769731/a505fb28-9c2a-11e6-8f27-ec80f23486b1.png)
is what I'm seeing when the ISO charset is single-quoted **or otherwise malformed** (such as extra spaces).

![double-quotes](https://cloud.githubusercontent.com/assets/2766036/19769770/d0cafad8-9c2a-11e6-8166-b3843d1f36fa.png)
is what I'm seeing when the charset is double-quoted **or when the charset is commented out and the page's encoding is manually set to ISO-8859-15**.

I get the correct emoji using the UTF-8 charset, and the same behavior as the first image when using the ASCII charset.

These tests were performed on Chrome 54, Windows 10.